dali/pipeline/operator/arg_helper.h

```diff
@@ -29,11 +30,17 @@ class ArgValue {
   ArgValue() = default;
   ArgValue(ArgValue &&) = default;
   inline ArgValue(const ArgValue &other) { *this = other; }
-  inline ArgValue(const std::string &name, const OpSpec &spec, ArgumentWorkspace *ws) {
-    if (spec.HasTensorArgument(name)) {
-      tensor_vector_ = &ws->ArgumentInput(name);
+  explicit inline ArgValue(const std::string &name) : name_(name) {
+  }
+
+  void Update(const OpSpec &spec, ArgumentWorkspace &ws, bool use_default = true) {
+    if (spec.HasTensorArgument(name_)) {
+      tensor_vector_ = &ws.ArgumentInput(name_);
+      gpu_dirty_ = true;
     } else {
-      value_ = spec.GetArgument<T>(name, ws);
+      is_set_ = spec.HasArgument(name_);
```

Comment:
Why not set tensor_vector_ to nullptr here?

Reply:
In real use cases it doesn't matter - arguments don't just change from tensor to non-tensor and back - but I can add it for some sort of generality/completeness.

Reply:
True